src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -452,3 +476,20 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
         )
         df._reader = self
         return df
+
+    def _upload_local_file_to_stage(self, path: str, format: str = "csv") -> str:
+        temp_stage = self._session.get_session_stage()
+        if os.path.exists(path) and os.path.isfile(path):
+            _ = self._session.file.put(path, temp_stage, auto_compress=False)
+            _, filename = os.path.split(path)
+            path = os.path.join(temp_stage, filename)
+        if os.path.exists(path) and os.path.isdir(path):
```

Comment:
hmm I just realized, we only support csv file here, right? So it should be 
```suggestion
        if os.path.exists(path) and os.path.isdir(path):
            if format != "csv":
                raise ValueError(f"Only support a local directory for reading CSV files, but got {format}")
```